rasa/shared/core/training_data/story_reader/story_step_builder.py

```diff
@@ -143,3 +142,26 @@ def _next_story_steps(self) -> List[StoryStep]:
             )
         ]
         return current_turns
+
+    def _generate_checkpoint_name_for_or_statement(
+        self, messages: List[UserUttered]
+    ) -> str:
+        """Generates a unique checkpoint name for an or statement.
+
+        The name is based on the current story/rule name,
+        the current place in the story,
+        and the involved intents/e2e messages.
+        """
+        messages_texts_or_intents = sorted([m.text or m.intent_name for m in messages])
+
+        current_steps_event_counts = [str(len(s.events)) for s in self.current_steps]
```

Comment:
Does this actually get the position in the story? Or is it just the position within the current checkpoint? i.e. of we had a story with two different OR blocks, both with the same possible intents, would the names still be globally unique?